eclair-node/src/main/scala/fr/acinq/eclair/api/handlers/Message.scala

```diff
@@ -39,5 +40,11 @@ trait Message {
     }
   }
 
-  val messageRoutes: Route = signMessage ~ verifyMessage
+  val sendOnionMessage: Route = postRequest("sendonionmessage") { implicit t =>
+    formFields("intermediateNodes".as[List[PublicKey]](pubkeyListUnmarshaller), "destination".as[PublicKey](publicKeyUnmarshaller)) { (intermediateNodes, destination) =>
```

Comment:
We should also take a hex-encoded payload for the recipient (which would be a tlv stream, not length-prefixed, that we should inject into the final payload). Probably no need to validate it at that point, it's for experimentation, so the caller should make sure they don't mess it up.

Reply:
The problem is that the final payload contains the `RouteBlindingEncryptedDataTlv`. If I let the user set this manually it will be completely unusable. The other solution is to duplicate the `buildMessage` function with some ugly ad-hoc code to build a TLV stream from a combination of type TLV and an untyped `ByteVector`.

Reply:
Have a look at what we do in `PaymentInitiator.scala` with `SendPaymentToNode`: we let users provide a list of `GenericTlv` that we insert in the tlv stream. I think it would work fine in our case, these should simply be added to the tlv stream of `OnionMessagePayloadTlv`, it shouldn't impact the blinded part AFAICT.

Reply:
Done